util/src/openssl_utils.cpp

```diff
@@ -0,0 +1,266 @@
+using std::string;
+using std::unique_ptr;
+using concord::util::crypto::KeyFormat;
+using concord::util::openssl_utils::UniquePKEY;
+using concord::util::openssl_utils::UniqueOpenSSLPKEYContext;
+using concord::util::openssl_utils::UniqueOpenSSLX509;
+using concord::util::openssl_utils::UniqueOpenSSLBIO;
+using concord::util::openssl_utils::OPENSSL_SUCCESS;
+using concord::util::openssl_utils::OPENSSL_FAILURE;
+using concord::util::openssl_utils::OPENSSL_ERROR;
+
+string CertificateUtils::generateSelfSignedCert(const string& origin_cert_path,
+                                                const string& public_key,
+                                                const string& signing_key) {
+  auto deleter = [](FILE* fp) {
+    if (nullptr != fp) {
+      fclose(fp);
+    }
+  };
+  unique_ptr<FILE, decltype(deleter)> fp(fopen(origin_cert_path.c_str(), "r"), deleter);
+  if (!fp) {
+    LOG_ERROR(OPENSSL_LOG, "Certificate file not found, path: " << origin_cert_path);
+    return string();
+  }
+
+  UniqueOpenSSLX509 cert(PEM_read_X509(fp.get(), nullptr, nullptr, nullptr));
+  if (!cert) {
+    LOG_ERROR(OPENSSL_LOG, "Cannot parse certificate, path: " << origin_cert_path);
+    return string();
+  }
+
+  UniquePKEY priv_key(EVP_PKEY_new());
+  UniqueOpenSSLBIO priv_bio(BIO_new(BIO_s_mem()));
+
+  if (BIO_write(priv_bio.get(), static_cast<const char*>(signing_key.c_str()), signing_key.size()) <= 0) {
+    LOG_ERROR(OPENSSL_LOG, "Unable to create private key object");
+    return string();
+  }
+
+  if (!PEM_read_bio_PrivateKey(priv_bio.get(), reinterpret_cast<EVP_PKEY**>(&priv_key), nullptr, nullptr)) {
+    LOG_ERROR(OPENSSL_LOG, "Unable to create private key object");
+    return string();
+  }
+  UniquePKEY pub_key(EVP_PKEY_new());
+  UniqueOpenSSLBIO pub_bio(BIO_new(BIO_s_mem()));
+
+  if (BIO_write(pub_bio.get(), static_cast<const char*>(public_key.c_str()), public_key.size()) <= 0) {
+    LOG_ERROR(OPENSSL_LOG, "Unable to write public key object");
+    return string();
+  }
+  if (!PEM_read_bio_PUBKEY(pub_bio.get(), reinterpret_cast<EVP_PKEY**>(&pub_key), nullptr, nullptr)) {
+    LOG_ERROR(OPENSSL_LOG, "Unable to create public key object");
+    return string();
+  }
+
+  if (OPENSSL_FAILURE == X509_set_pubkey(cert.get(), pub_key.get())) {
+    LOG_ERROR(OPENSSL_LOG, "Failed to set public key for certificate.");
+    return {};
+  }
+
+  if (OPENSSL_FAILURE == X509_sign(cert.get(), priv_key.get(), EVP_sha256())) {
+    LOG_ERROR(OPENSSL_LOG, "Failed to sign certificate using private key.");
+    return {};
+  }
+
+  UniqueOpenSSLBIO outbio(BIO_new(BIO_s_mem()));
+  if (!PEM_write_bio_X509(outbio.get(), cert.get())) {
+    LOG_ERROR(OPENSSL_LOG, "Unable to create certificate object");
+    return string();
+  }
+  string certStr;
+  int certLen = BIO_pending(outbio.get());
+  certStr.resize(certLen);
+  const auto res = BIO_read(outbio.get(), (void*)&(certStr.front()), certLen);
+  if (OPENSSL_FAILURE == res or OPENSSL_ERROR == res) {
```

Comment:
This is the only point in the core where the "or" keyword is used, except for your new io.cpp file.
This might be a personal preference, but for consistency I think you should change it to ||.

Reply:
Changed to `||` in https://github.com/arc-vmware/concord-bft/pull/6/commits/a20433b573177d6a2c80bd591ccfd0dba0f87600